var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -281,3 +283,50 @@ def setup_run_environment(self, env):
             env.prepend_path("HDF5_PLUGIN_PATH", os.path.dirname(all_libs[idx]))
         except ValueError:
             pass
+
+    @run_after("install")
+    def setup_install_tests(self):
+        """
+        Copy the example files after the package is installed to an
+        install test subdirectory for use during `spack test run`.
+        """
+        extra_install_tests = [join_path("testing", "install", "C")]
+        self.cache_extra_test_sources(extra_install_tests)
+
+    def test_run_executables(self):
+        """Run installed adios2 executables"""
+
+        commands_and_args = [("bpls", ["-v", "-V"])]
+
+        for cmd, opts in commands_and_args:
+            with test_part(
+                self, f"test_adios2_cmds_{cmd}", purpose=f"run installed adios2 executable {cmd}"
```

Comment:
Nit.  Simplification of naming/tracking embedded test parts is on my ToDo list but, in the meantime, I recommend using the name of the test part as the start of the names of embedded test parts so the results in CI are sorted "properly".  As it stands, the summary shows:

```
Adios2::test_build_example .. PASSED
Adios2::test_adios2_cmds_bpls .. PASSED
Adios2::test_run_executables .. PASSED
```

But if the names are based on the enclosing test part, it might be easier to infer the relation, e.g.:

```
Adios2::test_build_example .. PASSED
Adios2::test_run_executables .. PASSED
Adios2::test_run_executables_bpls .. PASSED
```

Though it may be more obvious in the CDash tables.

Reply:
Also, since you don't have more than one executable, you could just get rid of the embedded test part in this case and maybe call the test `test_run_bpls`.

Reply:
Sure, thanks.  I'll make these changes as well.

Reply:
I took your advice (I think) and renamed the tests and embedded parts so it should be more clear which bits are part of the same tests.  Let me know if I missed the mark though.  Here's the output now:

```console
Adios2::test_examples_build .. PASSED
Adios2::test_examples_run_adios_c_mpi_test .. PASSED
Adios2::test_examples_run_adios_adios2c_test .. PASSED
Adios2::test_examples_run_adios_c_test .. PASSED
Adios2::test_examples .. PASSED
Adios2::test_run_executables_bpls .. PASSED
Adios2::test_run_executables_adios2-config .. PASSED
Adios2::test_run_executables .. PASSED
```